docs/user/readme.adoc

```diff
@@ -183,11 +183,20 @@ Once `neovim/nvim-lsp` is installed, use `+lua require'nvim_lsp'.rust_analyzer.s
 
 === Sublime Text 3
 
-Prerequisites:
+Prerequisites: You have installed the <<language-server-binary,`rust-analyzer` binary>>.
+
+You also need the `LSP` package. To install it:
+
+1. If you've never installed a Sublime Text package, install Package Control:
+   * Open the command palette (Win/Linux: `ctrl+shift+p`, Mac: `cmd+shift+p`)
+   * Type `Install Package Control`, press enter
+2. In the command palette, run `Package control: Install package`, and in the list that pops up, type `LSP` and press enter.
+
+Finally, with your Rust project open, in the command palette, run `LSP: Enable Language Server In Project` or `LSP: Enable Language Server Globally`, then select `rust-analyzer` in the list that pops up enable to enable the rust-analyzer LSP. The latter means that rust-analzyer is enabled by default in Rust projects.
 
-`LSP` package.
+If it worked, you should see "rust-analzyer, Line X, Column Y" on the left side of the bottom bar, and after waiting a bit, functionality like tooltips on hovering over variables should become available.
 
-Invoke the command palette (`ctrl+shift+p`) and type LSP enable to locally/globally enable the rust-analyzer LSP (type LSP enable, then choose either locally or globally, then select rust-analyzer)
+If you get an error saying `No such file or directory: 'rust-analyzer'` even though the binary is on your `$PATH`, there is likely a problem where Sublime doesn't see the same `$PATH` as your shell, see https://github.com/rust-analyzer/rust-analyzer/issues/1811[this issue]. On Unix, if you installed Rust with `rustup`, moving the binary to `$HOME/.cargo/bin` should help.
```

Comment:
```suggestion
If you get an error saying `No such file or directory: 'rust-analyzer'` even though the binary is on your `$PATH`, the likely explanation is that Sublime doesn't see the same `$PATH` as the shell, see https://github.com/rust-analyzer/rust-analyzer/issues/1811[this issue]. On Unix, running the editor from a shell or changing the `.desktop` file to set the environment, should help.
```

Reply:
Running the editor from the shell didn't help for me (on Mac). If other people had success with that, it's fine though.

Reply:
My understanding of the issue is that the `$PATH` customization only applies to shells (or possibly login shells, depending on where it's made), while the desktop environment doesn't run one. This affects Unix DEs and MacOS, but isn't specific to Sublime, so we should probably move the remark somewhere above.

Reply:
I tried again and running the editor from the shell works (maybe before the process was still running). Looking at the env vars of the running process confirms that. I can move this tip up to the section about installing the binary. Should we add as a third option modifying ~/.profile and logging in again?

Reply:
> I can move this tip up to the section about installing the binary.

That would be swell :-).

> Should we add as a third option modifying ~/.profile and logging in again?

Well, that's the issue, adding it in `~/.profile` doesn't work (it doesn't get executed). That's why I was recommending `~/.pam_environment` in the linked issue.

Reply:
Hmm, then I don't understand yet why moving the binary to `~/.cargo/bin` helped, when that dir is added to the `PATH` in `~/.profile` (and nowhere else). Unfortunately I can't log out of my machine at the moment to check whether it would work. 

I just found out [`~/.profile` doesn't get executed if you have a `.bash_profile`](https://github.com/rust-lang/rustup/blob/2138b7ab0f1855f45127e5a42eea902b8935ba4e/src/cli/self_update.rs#L1141-L1142), could that explain it?

Reply:
> then I don't understand yet why moving the binary to ~/.cargo/bin helped

I'm not sure. At the time of the linked issue, `~/.cargo/bin` was the default install path anyway.

> I just found out ~/.profile doesn't get executed if you have a .bash_profile, could that explain it?

That really depends on the system. ~/.profile` being executed executed depends on having a line like `if [ -r ~/.profile ]; then . ~/.profile; fi` in your `~/.bash_profile`. The `rustup` comment talks about the case where `~/.bash_profile` doesn't exist yet.

Reply:
True ... that's confusing.